x-pack/plugins/uptime/server/lib/requests/get_monitor_states.ts

```diff
@@ -4,37 +4,24 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { CONTEXT_DEFAULTS, QUERY } from '../../../common/constants';
 import { UMElasticsearchQueryFn } from '../adapters';
-import { SortOrder, CursorDirection, MonitorSummariesResult } from '../../../common/runtime_types';
-import { QueryContext, MonitorSummaryIterator } from './search';
+import { MonitorSummariesResult } from '../../../common/runtime_types';
+import { QueryContext } from './search';
 import { HistogramPoint, Histogram } from '../../../common/runtime_types';
-import { getHistogramInterval } from '../helper/get_histogram_interval';
-
-export interface CursorPagination {
-  cursorKey?: any;
-  cursorDirection: CursorDirection;
-  sortOrder: SortOrder;
-}
+import { fetchChunk } from './search/fetch_chunk';
+import { MonitorSummaryIterator } from './search/monitor_summary_iterator';
 
 export interface GetMonitorStatesParams {
   dateRangeStart: string;
   dateRangeEnd: string;
-  pagination?: CursorPagination;
   pageSize: number;
   filters?: string | null;
   statusFilter?: string;
+  pageIndex: number;
```

Comment:
As we discussed, I think we should stick with prev/next pagination. Page based pagination doesn't perform in (relatively) constant time as the current approach does.

I'd be willing to change my mind if there's benchmarks to back-up the change.